submissions/api.py

```diff
@@ -988,7 +988,21 @@ def _get_or_create_student_item(student_item_dict):
                     }
                 )
                 raise SubmissionRequestError(field_errors=student_item_serializer.errors) from student_error
-            return student_item_serializer.save()
+            try:
+                # This is required because we currently have automatic request transactions turned on in the LMS.
+                # Database errors mess up the "atomic" block so we have to "insulate" against them with an
+                # inner atomic block (https://docs.djangoproject.com/en/4.0/topics/db/transactions/)
+                with transaction.atomic():
+                    return student_item_serializer.save()
+            except IntegrityError:
+                # In the case where a student item does not exist and multiple calls to this function happen, there
+                # can be a race condition where the first get has no results, but once the save happens there is already
+                # a version of this student item. In that case, try just loading again and see if the item exists now.
+                try:
+                    return StudentItem.objects.get(**student_item_dict)
+                except StudentItem.DoesNotExist:
+                    pass
+                raise
```

Comment:
Should this raise... something in particular?

Reply:
raise without arguments will raise the "current" exception, in this case, the IntegrityError. Being so close to the StudentItem.DoesNotExist, though? maybe i should be explicit for readability